pages/[...slug].tsx

```diff
@@ -141,63 +138,70 @@ const DocSlugs = ({ source, frontMatter, pagination }: Props) => {
 export default DocSlugs;
 
 export const getStaticProps = async ({ params }) => {
-    // Absolute path of the docs file
-    const postFilePath = path.join(DOCS_PATH, `${path.join(...params.slug)}.mdx`);
-    // Raw Mdx File Data Buffer
-    const source = fs.readFileSync(postFilePath);
-
-    /**
-     * Content: Mdx Data
-     * data: FrontMatter Data
-     */
-    const { content, data } = matter(source);
-
-    const allDocs = getAllDocs();
-    const nav = getNavfromDocs(allDocs);
-    const [currentDocSlug] = params.slug as string[];
-    const currentDocs = allDocs.filter((doc) => doc.url.includes(`/${currentDocSlug}/`));
-    const { previousPost, nextPost } = getPagination(currentDocs, params.slug as string[]);
-    const pagination = { previousPost, nextPost };
-    const toc = [];
-    const mdxSource = await renderToString(content, {
-        components,
-        // Optionally pass remark/rehype plugins
-        mdxOptions: {
-            remarkPlugins: [
-                require('@/lib/remark-code-header'),
-                require('@fec/remark-a11y-emoji'),
-                withTableofContents(toc),
-                imagePlugin
-            ],
-            rehypePlugins: [mdxPrism]
-        },
-        scope: data
-    });
-    return {
-        props: {
-            toc,
-            pagination,
-            nav: { [currentDocSlug]: nav[currentDocSlug] },
-            source: mdxSource, // { compiledSource: mdxSource.compiledSource },
-            frontMatter: data,
+    try {
+        if (params.slug.join('/') !== params.slug.join('/').toLowerCase())
+            return {
+                redirect: {
+                    destination: `/${params.slug.join('/').toLowerCase()}`,
```

Comment:
should we check if we can do this at deploy time and not run time ? 

also should this be a middleware and not part of page ?